packages/aws-cdk-lib/aws-eks/lib/service-account.ts

```diff
@@ -91,19 +163,61 @@ export class ServiceAccount extends Construct implements IPrincipal {
-    */
-    const conditions = new CfnJson(this, 'ConditionJson', {
-      value: {
-        [`${cluster.openIdConnectProvider.openIdConnectProviderIssuer}:aud`]: 'sts.amazonaws.com',
-        [`${cluster.openIdConnectProvider.openIdConnectProviderIssuer}:sub`]: `system:serviceaccount:${this.serviceAccountNamespace}:${this.serviceAccountName}`,
-      },
-    });
-    const principal = new OpenIdConnectPrincipal(cluster.openIdConnectProvider).withConditions({
-      StringEquals: conditions,
-    });
-    this.role = new Role(this, 'Role', { assumedBy: principal });
+    let principal: IPrincipal;
+    if (props.identityType !== IdentityType.POD_IDENTITY) {
+      /* Add conditions to the role to improve security. This prevents other pods in the same namespace to assume the role.
+      * See documentation: https://docs.aws.amazon.com/eks/latest/userguide/create-service-account-iam-policy-and-role.html
+      */
+      const conditions = new CfnJson(this, 'ConditionJson', {
+        value: {
+          [`${cluster.openIdConnectProvider.openIdConnectProviderIssuer}:aud`]: 'sts.amazonaws.com',
+          [`${cluster.openIdConnectProvider.openIdConnectProviderIssuer}:sub`]: `system:serviceaccount:${this.serviceAccountNamespace}:${this.serviceAccountName}`,
+        },
+      });
+      principal = new OpenIdConnectPrincipal(cluster.openIdConnectProvider).withConditions({
+        StringEquals: conditions,
+      });
+
+    } else {
+      /**
+       * Identity type is POD_IDENTITY.
+       * Create a service principal with "Service": "pods.eks.amazonaws.com"
+       * See https://docs.aws.amazon.com/eks/latest/userguide/pod-identities.html
+       */
+
+      // EKS Pod Identity does not support Fargate
+      if (cluster instanceof FargateCluster) {
+        throw Error('Pod Identity is not supported in Fargate. Use IRSA identity type instead.');
+      }
+      principal = new ServicePrincipal('pods.eks.amazonaws.com');
+    }
+
+    const role = new Role(this, 'Role', { assumedBy: principal });
+
+    // pod identities requires 'sts:TagSession' in its principal actions
+    if (props.identityType === IdentityType.POD_IDENTITY) {
+      /**
+       * EKS Pod Identities requires both assumed role actions otherwise it would fail.
+       */
+      role.assumeRolePolicy!.addStatements(new PolicyStatement({
+        actions: ['sts:AssumeRole', 'sts:TagSession'],
+        principals: [new ServicePrincipal('pods.eks.amazonaws.com')],
+      }));
+
+      // ensure the pod identity agent
+      cluster.eksPodIdentityAgent;
+
+      // associate this service account with the pod role we just created for the cluster
+      new CfnPodIdentityAssociation(this, 'Association', {
+        clusterName: cluster.clusterName,
+        namespace: props.namespace ?? 'default',
+        roleArn: role.roleArn,
+        serviceAccount: this.serviceAccountName,
+      });
```

Comment:
Should we pass in props like `annotations`, `labels`, etc. I don't see them being used otherwise.

Reply:
annotatoins and labels will be used here when we create the ServiceAccount manifest.

https://github.com/aws/aws-cdk/blob/8ef51b79e5573fc9cfc1abc4f35912a155bfe633/packages/aws-cdk-lib/aws-eks/lib/service-account.ts#L238-L245